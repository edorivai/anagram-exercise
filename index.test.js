const { createIndex, match } = require('./index');

describe('#createIndex', () => {
	test('should create a simple index', async () => {
		const result = createIndex('./fixtures/sample.txt');
		expect(result).toMatchSnapshot();
	});
});

describe('#match', () => {
	test('should find items that are indexed', async () => {
		const index = require('./fixtures/index.json');
		expect(match('edo', index)).toMatchSnapshot();
	});
	
	test('should be case insensitive', async () => {
		const index = require('./fixtures/index.json');
		const result = match('EDO', index);
		expect(result).not.toBe(undefined);
		expect(result).toMatchSnapshot();
	});
	
	test('should return undefined if the input is not found in the dictionary', async () => {
		const index = require('./fixtures/index.json');
		expect(match('not-in-dictionary', index)).toBe(undefined);
	});
});