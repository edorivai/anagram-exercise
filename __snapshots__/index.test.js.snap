// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createIndex should create a simple index 1`] = `
Object {
  "Bac": Array [
    "Bca",
  ],
  "aaagmnr": Array [
    "anagram",
    "nagaram",
  ],
  "abc": Array [
    "abc",
    "bca",
  ],
  "deo": Array [
    "edo",
    "doe",
  ],
}
`;

exports[`#match should be case insensitive 1`] = `
Array [
  "edo",
  "doe",
]
`;

exports[`#match should find items that are indexed 1`] = `
Array [
  "edo",
  "doe",
]
`;
